synapse/events/utils.py

```diff
@@ -171,6 +171,18 @@ def add_fields(*fields: str) -> None:
     elif room_version.msc2716_redactions and event_type == EventTypes.MSC2716_MARKER:
         add_fields(EventContentFields.MSC2716_INSERTION_EVENT_REFERENCE)
 
+    # Protect the rel_type and event_id fields under the m.relates_to field.
+    if room_version.msc3389_relation_redactions:
+        relates_to = event_dict["content"].get("m.relates_to")
```

Comment:
It looks like we aren't using the default value from the `.get(..., default)` call and the mixing of getters here seems odd. Alternative: `event_dict["content"]["m.relates_to"]`

This is more of a nothing/doesn't matter lint sort of situation but just curious on what we try to prefer.

Reply:
`m.relates_to` might not exist which is why we need to use `get(...)`.

The bare `relates_to` check will filter out the implicit `None` that is the default value. (Although honestly that isn't needed since `None` isn't a `Mapping`.)

Reply:
Ahh, I see. I assumed we wouldn't see the `KeyError` (JavaScript mind mush) but `event_dict["content"]["m.relates_to"]` -> `KeyError: 'm.relates_to'`

Reply:
Yeah, exactly. 👍  I'll make the minor change to not check that case specifically. (Note that in this case we can treat the field missing the same as if the field is set to `null` -- either case is invalid in the same way, so we don't care. Otherwise we'd have to first guard the entire section with `"m.relates_to" in event["content"]`)